csharp/src/Drivers/BigQuery/BigQueryStatement.cs

```diff
@@ -349,15 +374,27 @@ sealed class ReadRowsStream : Stream
             ReadOnlyMemory<byte> currentBuffer;
             bool first;
             int position;
+            bool hasRows = true;
```

Comment:
Consider moving both initialization cases into ctor instead of splitting between the two.